pkg/ruler/evaluator_remote_test.go

```diff
@@ -411,6 +414,77 @@ func TestRemoteEvalUnsupportedResultResponse(t *testing.T) {
 	require.ErrorContains(t, err, fmt.Sprintf("unsupported result type: %q", loghttp.ResultTypeStream))
 }
 
+// TestRemoteEvalQueryTags validates that proper query tags are passed
+func TestRemoteEvalQueryTags(t *testing.T) {
+	defaultLimits := defaultLimitsTestConfig()
+	limits, err := validation.NewOverrides(defaultLimits, nil)
+	require.NoError(t, err)
+
+	now := time.Now()
+
+	cli := mockClient{
+		handleFn: func(ctx context.Context, in *httpgrpc.HTTPRequest, opts ...grpc.CallOption) (*httpgrpc.HTTPResponse, error) {
+			var queryTags string
+			for _, h := range in.Headers {
+				if h.Key == string(httpreq.QueryTagsHTTPHeader) {
+					queryTags = strings.Join(h.Values, ",")
+				}
+			}
+			require.Equal(t, "ruler,Source=ruler,Rule=foo,Kind=alerting", queryTags)
+
+			// this is somewhat bleeding the abstraction, but it's more idiomatic/readable than constructing
+			// the expected JSON response by hand
+			resp := loghttp.QueryResponse{
```

Comment:
Why do we need a valid response? Aren't we just looking at the headers in the test?